### PR TITLE
HAI-2231 Add code for migrating a single hanke

### PR DESCRIPTION
**Pull request by corvidian**
# Description
Add code for migrating a single hanke.

Still needs:
- Logging
- Starting the migration at startup or with a timer

## Type of change

- [ ] Bug fix 
- [X] New feature 
- [ ] Other

# Instructions for testing
Until there's a trigger, this can only be tested with integration and unit tests-

# Checklist:

- [X] I have written new tests (if applicable)
- [X] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
 or other location: 

# Other relevant info
I'll add logging in a separate PR since this is getting very long.
The trigger for running the migration, which loads all hanke IDs and migrates them one-by-one will be added later.

### CODE REVIEW

File: services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationServiceTest.kt [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+                    ApplicationFactory.createContact(
+                        firstName = "Hannu",
+                        lastName = null,
+                        email = " "
+                    )
+                )
+
+            val response = contacts.contactWithLeastMissingFields()
+
+            assertThat(response).isNotNull().isSameInstanceAs(contacts[1])
```
Comment (umeetiusbaar) [inline services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationServiceTest.kt:117]: So, it is Pekka? Reading just this test doesn't make it very clear how we have came into that conclusion. The names have some priority over phone? None of the contacts have email address, should there be such a test object too?

File: services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+            val contacts = arrayOf(nonOrderer, missingTwo, firstMissingOne, secondMissingOne)
+            val data =
+                ApplicationFactory.createCableReportApplicationData(
+                    customerWithContacts =
+                        ApplicationFactory.createCompanyCustomer().withContacts(*contacts),
+                )
+
+            val result = HakemusMigrationService.findOrderer(data)
+
+            assertThat(result).isSameInstanceAs(firstMissingOne)
```
Comment (umeetiusbaar) [inline services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationServiceTest.kt:179]: This is more clear than the previous test with the differences between test objects.

File: services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationServiceTest.kt (hunk 1 above)
Comment (umeetiusbaar) [reply services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationServiceTest.kt:117]: Well, nevermid, this is not so important and you can ignore this. I thought that starting reading from the tests would make the logic more clear but it turned out to be the other way round.

**Review by umeetiusbaar — APPROVED** [review]

File: services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationServiceTest.kt (hunk 1 above)
Comment (corvidian) [reply services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationServiceTest.kt]: I changed the test to create a `Contact` directly, without using the factory method. I think this makes it easier to see which fields are missing and which are not. I also made the contacts named values, to give them some context.

File: services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationServiceTest.kt (hunk 2 above)
Comment (corvidian) [reply services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationServiceTest.kt:191]: I changed the test to create a `Contact` directly, without using the factory method. I think this makes it easier to see which fields are missing and which are not.

File: services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationServiceTest.kt (hunk 1 above)
Comment (umeetiusbaar) [reply services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationServiceTest.kt]: Thank you!

**Review by umeetiusbaar — APPROVED** [review]

**Pull request merged**